Quick-Start/Anatomy_of_extension.md

```diff
@@ -1,418 +1,418 @@
+
+The code file plays a crucial role in the development of a PostgreSQL extension as it implements the logic and behaviour of the extension's functionality. It allows developers to extend the capabilities of PostgreSQL by adding custom functionality that is not available in the core database system. 
+
+The code file interacts directly with the PostgreSQL server through the PostgreSQL Extension API, which provides functions and macros for defining extension objects, handling queries, and accessing database resources. 
+
+Here's a simplified example of a code file (`my_extension.c`) for a PostgreSQL extension: 
+
+```c 
+#include "postgres.h" 
+#include "fmgr.h" 
+ 
+PG_MODULE_MAGIC; 
+
+// Define the extension's functions 
+PG_FUNCTION_INFO_V1(my_extension_func); 
+ 
+// Implementation of the extension's function  
+Datum my_extension_func(PG_FUNCTION_ARGS) 
+{ 
+    // Retrieve the function arguments 
+    int arg1 = PG_GETARG_INT32(0); 
+    int arg2 = PG_GETARG_INT32(1); 
+     
+    // Perform some operation 
+    int result = arg1 + arg2; 
+     
+    // Return the result 
+    PG_RETURN_INT32(result); 
+} 
+ 
+``` 
+ 
+In this example: 
+-   The code file includes necessary header files (`postgres.h` and `fmgr.h`) that provide access to the PostgreSQL Extension API. 
+-   `PG_MODULE_MAGIC` is a macro that indicates the compatibility version of the PostgreSQL Extension API. It must be included in the code file. 
+-   `PG_FUNCTION_INFO_V1` is a macro that defines the information for an extension function. It is used to declare the function to the PostgreSQL server. 
+-   `my_extension_func` is the implementation of the extension's function. It takes two integer arguments, adds them together, and returns the result. 
+-   `PG_GETARG_INT32` is a macro that retrieves the function arguments of type `int32` (integer).   
+-   `PG_RETURN_INT32` is a macro that sets the return value of the function to an `int32` value. 
+    
+The code file can include additional functions, data types, operators, and other database entities as required by the extension's functionality. It provides the foundation for extending PostgreSQL and allows developers to define custom behaviour that can be executed within the database. 
+ 
+## Creating Code Files for PostgreSQL Extensions: 
+Code Files: The code files contain the actual implementation of the extension's functionality. These files are typically written in a programming language such as C, C++, Python, or any other supported language. The code files interact with the database through the SQL interface and provide additional features and capabilities. Create a quick start on how to go about writing Code Files (how it will help the developer)while creating an Extension in Postgres. 
```

Comment:
Non-SQL source code does not use the SQL interface, but the Postgres APIs such as the macro examples above. It gets compiled into a shared object library that you can then reference from the SQL script.